daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -463,6 +463,19 @@ private[lf] object SBuiltin {
     }
   }
 
+  // Check if the supplied text is valid as a UserId.
+  // Returns None if the text is valid, or Some(msg) if the text is invalid,
+  // where msg is a human readable message explaining why the text is not valid.
+  final case object SBValidateTextAsUserId extends SBuiltinPure(1) {
```

Comment:
Let’s please make this crash in on ledger mode just to be extra safe.

Maybe also worth testing in `SBuiltinTest` but I’m also happy with just having the script tests for this.